tests/core/test_nlg.py

```diff
@@ -328,3 +331,271 @@ def test_nlg_fill_response_quick_replies(
         filled_slots={quick_replies_slot_name: quick_replies_slot_value},
     )
     assert result == {"quick_replies": str(quick_replies_slot_value)}
+
+
+@pytest.mark.parametrize(
+    ("slot_name", "slot_value", "output_channel", "response_variation"),
+    (
+        ("slot test", "A", "os", "Conditional Response A"),
+        ("slot test", "A", "app", "Conditional app response A"),
+        ("slot test", "B", "", "Conditional Response B"),
+        ("slot test", "C", "", "Default response"),
+    ),
+)
+async def test_nlg_conditional_response_variations(
+    slot_name, slot_value, output_channel, response_variation
+):
+    responses = {
+        "utter_test": [
+            {
+                "text": "Conditional Response A",
+                "condition": [{"type": "slot", "name": "slot test", "value": "A"}],
+                "channel": "os",
+            },
+            {
+                "text": "Conditional app response A",
+                "condition": [{"type": "slot", "name": "slot test", "value": "A"}],
+                "channel": "app",
+            },
+            {
+                "text": "Conditional Response B",
+                "condition": [{"type": "slot", "name": "slot test", "value": "B"}],
+            },
+            {"text": "Default response"},
+        ]
+    }
+    t = TemplatedNaturalLanguageGenerator(responses=responses)
+
+    slot = Slot(name=slot_name, initial_value=slot_value, influence_conversation=False)
+    tracker = DialogueStateTracker(sender_id="nlg_test", slots=[slot])
+
+    r = await t.generate(
+        utter_action="utter_test", tracker=tracker, output_channel=output_channel
+    )
+    assert r.get("text") == response_variation
+
+    no_slots_tracker = DialogueStateTracker(sender_id="nlg_test_default", slots=None)
+    default_response = await t.generate(
+        utter_action="utter_test", tracker=no_slots_tracker, output_channel=""
+    )
+    assert default_response.get("text") == "Default response"
```

Comment:
optional: slightly nitpicking but in my opinion that would be a different test. Not a huge deal though (hence `optional`)